content/ko/docs/tasks/administer-cluster/safely-drain-node.md

```diff
@@ -0,0 +1,110 @@
+---
+# reviewers:
+# - davidopp
+# - mml
+# - foxish
+# - kow3ns
+title: 안전하게 노드 드레인하기
+content_type: task
+min-kubernetes-server-version: 1.5
+---
+
+<!-- overview -->
+이 페이지는 사용자가 정의한 PodDisruptionBudget을 선택적으로 준수하면서
+{{< glossary_tooltip text="노드" term_id="node" >}}를 안전하게 드레인하는 방법을 보여준다.
+
+## {{% heading "prerequisites" %}}
+
+{{% version-check %}}
+또한 이 작업은 다음 사전 요구 사항을 충족한다고 가정한다.
+  1. 노드를 드레인하는 동안 애플리케이션에 고가용성이 필요하지
+     않거나, 혹은
+  1. [PodDisruptionBudget](/ko/docs/concepts/workloads/pods/disruptions/) 개념에 대해 읽었으며,
+     이를 필요로 하는 애플리케이션을 위해
+     [설정된 PodDisruptionBudgets](/docs/tasks/run-application/configure-pdb/)을 가지고 있다.
+
+<!-- steps -->
+
+## (선택 사항) 중단(disruption) 버짓 구성하기 {#configure-poddisruptionbudget}
+
+유지보수 중에 워크로드를 계속 사용할 수 있도록,
+[PodDisruptionBudget](/ko/docs/concepts/workloads/pods/disruptions/)를 구성할 수 있다.
+
+만약 드레인하려는 노드에서 애플리케이션이 실행중이거나 실행될 수 있는
+가용성이 중요하다면, 먼저 [configure a PodDisruptionBudgets](/docs/tasks/run-application/configure-pdb/)을
+수행한 뒤에 이 가이드를 따른다.
+
+## `kubectl drain`을 사용하여 서비스에서 노드 제거하기
+
+노드에서 유지보수(예로 커널 업그레이드, 하드웨어 유지보수 등)을 수행하기 전에
+`kubectl drain`을 사용하여 노드에서 모든 파드를 안전하게 축출할 수 있다.
+안전한 축출은 파드의 컨테이너가
+[정상적으로 종료(gracefully terminate)](/ko/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)될 수 있도록 하며,
+사용자가 지정한 PodDisruptionBudgets을 준수한다.
+
+{{< note >}}
+기본적으로 `kubectl drain`은 종료할 수 없는 노드의
+특정 시스템 파드는 무시한다. 자세한 내용은
+[kubectl drain](/docs/reference/generated/kubectl/kubectl-commands/#drain)
+문서를 참고한다.
+{{< /note >}}
+
+`kubectl drain`이 성공적으로 반환한 경우, 모든 파드
+(이전 단락에서 설명한 무시된 파드를 제외)가 (사용자가 원하는 유예 종료 기간을
+준수하고 사용자가 정의한 PodDisruptionBudget을 준수하여) 안전하게 축출되었음을
+나타낸다. 그런 다음 물리적 머신의 전원을 끄거나
+클라우드 플랫폼에서 실행 중인 경우 가상 머신을 삭제하여
+노드를 끄는 것이 안전하다.
+
+먼저 축출하려는 노드의 이름을 식별한다. 클러스터의 모든 노드는 다음과 같이 나열할 수 있다.
+
+```shell
+kubectl get nodes
+```
+
+그런 다음, 쿠버네티스에 노드를 비우라고 지시한다.
+
+```shell
+kubectl drain <node name>
+```
+
+(오류가 발생하지 않고) 반환되고 나면, 노드의 전원을 끄거나
+(또는 클라우드 플랫폼의 경우 노드를 지원하는 가상 머신을 삭제한다).
+만약 유지보수 작업 중에 노드를 클러스터에 남겨두려면 다음을 실행한다.
+
+```shell
+kubectl uncordon <node name>
+```
+다음에 쿠버네티스에게 노드에 새로운 파드 스케줄링을 재개할 수 있음을 알려준다.
```

Comment:
[원문]
> Once it returns (without giving an error), you can power down the node (or equivalently, if on a cloud platform, delete the virtual machine backing the node). If you leave the node in the cluster during the maintenance operation, you need to run
```
kubectl uncordon <node name>
```
> afterwards to tell Kubernetes that it can resume scheduling new pods onto the node.

- 명령어 앞뒤의 문장은 서로 이어져 있는 하나의 문장으로 보입니다.
- 'returns' 는 직역하면 '반환' 이지만, 문맥 상 '위쪽 명령어 실행 결과가 반환되면' 의 뜻으로 보이며,
  좀 더 이해가 쉽게 의역을 할 수도 있겠습니다.

이러한 사항 등을 고려하여 조금 다듬어 보았습니다.

```suggestion
위의 명령이 (오류 없이) 실행 완료되면, 노드의 전원을 끌 수 있다
(또는, 클라우드 플랫폼의 경우, 해당 노드로서 기능했던 가상 머신을 삭제할 수 있다).
만약 유지보수 작업 중에 노드를 클러스터에 남겨두려면 다음을 실행한다.

```shell
kubectl uncordon <node name>
```
L78: "이렇게 함으로써 노드에 새로운 파드 스케줄링을 재개할 수 있음을 쿠버네티스에게 알려줄 수 있다."
```

Reply:
자연스러운 의역에 좀 더 신경 써보겠습니다. 리뷰 감사합니다! 적용하겠습니다.